cpp/src/parquet/encoding.cc

```diff
@@ -2466,6 +2468,99 @@ class DeltaByteArrayDecoder : public DecoderImpl,
 // ----------------------------------------------------------------------
 // BYTE_STREAM_SPLIT
 
+namespace {
+template <typename T, bool has_sse2_or_higher>
+struct ByteStreamSplitDecodePath {
+  static int Decode(T* buffer, int max_values, int num_values_in_buffer,
+                    const uint8_t* data, int& num_values_decoded, int& len_remaining) {
+    throw ParquetException("This should be unreachable.");
+  }
```

Comment:
Instead of this invalid execution path, just don't implement this and only declare the function here?

Reply:
No longer relevant since the code was changed.